src/main/haskell/kore/src/Kore/Step/Substitution.hs

```diff
@@ -92,22 +70,66 @@ mergeSubstitutions tools first second = do
     -- TODO(virgil): Return the actual condition here.
     return (makeTruePredicate, substitution, proof)
 
+-- | Merge and normalize two unification substitutions
+mergeAndNormalizeSubstitutions
+    ::  ( MetaOrObject level
+        , Ord (variable level)
+        , OrdMetaOrObject variable
+        , SortedVariable variable
+        , Show (variable level)
+        , IntVariable variable
+        , Hashable variable
+        )
+    => MetadataTools level StepperAttributes
+    -> UnificationSubstitution level variable
+    -> UnificationSubstitution level variable
+    -> Either
+        ( UnificationOrSubstitutionError     level variable )
+        ( IntCounter
+          ( PredicateSubstitution level variable
+          , UnificationProof level variable
+          )
+        )
+mergeAndNormalizeSubstitutions tools first second = do
```

Comment:
The nested `Monad` makes this code difficult to parse, but a much simpler implementation of this function is
```.hs
mergeAndNormalizeSubstitutions tools first second = do
    (substitutionList, proof) <-
        normalizeSubstitutionDuplication' (first ++ second)
    predSubstitution <- normalizeSubstitution' substitutionList
    -- TODO(virgil): Return the actual condition here. and proofs
    return $ (,) <$> predSubstitution <*> pure proof  -- Inside IntCounter
```

Reply:
Oh wow, I didn't look at rewriting this after I removed the mergeConditions section. you are right, much simpler...